src/core/server/csp/config.ts

```diff
@@ -7,28 +7,56 @@
  */
 
 import { TypeOf, schema } from '@kbn/config-schema';
+import { ServiceConfigDescriptor } from '../internal_types';
+
+const configSchema = schema.object(
+  {
+    rules: schema.maybe(schema.arrayOf(schema.string())),
+    script_src: schema.arrayOf(schema.string(), { defaultValue: [] }),
```

Comment:
I did some manual testing, and using this config causes all users to see the "Please upgrade your browser" page.:
```yml
csp.script_src: ["'unsafe-inline'"]
```

That's because our "strict" check assumes that clients should not be able to execute a block of inline JavaScript.

Maybe we need to check for that token and throw an error if it is used unless `csp.strict: false` is also configured?

Reply:
> I did some manual testing, and using this config causes all users to see the "Please upgrade your browser" 

I guess this can also be achieved via `csp.rules`? If so, I would probably open a distinct issue for it, WDYT?

Reply:
> I guess this can also be achieved via `csp.rules`? If so, I would probably open a distinct issue for it, WDYT?

Yes it can be, but 1. that is deprecated, and 2. we don't really have the ability to easily parse `csp.rules` like we do `csp.script_src`. I'd say that if someone is using `csp.rules` to write a raw CSP, they need to know what they're doing, but if they're using our new config properties then we should make a best effort to include "guard rails" to prevent simple foot-guns, IMO.

Reply:
See https://github.com/elastic/kibana/pull/102059#issuecomment-862081807 for a more global question